src/content/peerconnection/bandwidth/js/main.js

```diff
@@ -288,3 +314,43 @@ window.setInterval(() => {
     lastResult = res;
   });
 }, 1000);
+
+function syntheticVideoStream({width = 640, height = 480, signal} = {}) {
+  const canvas = Object.assign(
+      document.createElement('canvas'), {width, height}
+  );
+  const ctx = canvas.getContext('2d');
+  const stream = canvas.captureStream();
+
+  let count = 0;
+  setInterval(() => {
+    // Use relatively-prime multipliers to get a color roll
+    const r = Math.abs((count*2)%512 - 256);
+    const g = Math.abs((count*3)%512 - 256);
+    const b = Math.abs((count*5)%512 - 256);
+    ctx.fillStyle = `rgb(${r}, ${g}, ${b})`;
```

Comment:
rgb accepts numbers between 0 and 255 (incl) or percentages from 0% to 100%.

Reply:
The purpose of the fancy math is to create a signal that changes enough to give the codec something to do, but without to many bandwidth spikes. I may iterate once more over this.

Reply:
Ok, so is it intentional to pass negative numbers to rgb? I assume they would be coerced to zero, but I'm not sure.